File: yumyum-back/src/main/java/com/yumyum/domain/user/dto/SignUpRequest.java
package com.yumyum.domain.user.dto;

import com.yumyum.domain.user.entity.User;
import lombok.Data;
import lombok.ToString;

import javax.validation.Valid;
import java.time.LocalDateTime;

@Data
@Valid
@ToString
public class SignUpRequest {

    private String email;

    private String nickname;

    private String introduction;

    private String profilePath;

    public User toEntity(final LocalDateTime nowTime){
        return User.builder()
                .email(email)
                .nickname(nickname)
                .introduction(introduction)
                .profilePath(profilePath)
                .createdDate(nowTime)
                .modifiedDate(nowTime)
                .build();
    }
}

File: AOS/app/src/main/java/com/omnyom/yumyum/model/signup/SignUpRequest.kt
package com.omnyom.yumyum.model.signup

class SignUpRequest(private val email: String,
                    private val nickname: String,
                    private val introduction: String,
                    private val profilePath: String) {
    fun get() : HashMap<String, String> {
        return hashMapOf(
                "email" to email,
                "nickname" to nickname,
                "introduction" to introduction,
                "profilePath" to profilePath
        )
    }
}
